core/net/src/hybrid/client.rs

```diff
@@ -134,6 +136,29 @@ proxy!(
     sockets,
     |client: Client| { futures::future::ready(client.sockets()) }
 );
+proxy!(
+    FindNode(NodeId) -> anyhow::Result<FindNodeResponse>,
+    find_node,
+    |client: Client, msg: FindNode| async move {
+        let res = client.find_node(msg.0).await?;
+        let identities = res.identities.into_iter()
+            .map(|i| NodeId::try_from(&i.node_id.to_vec()))
+            .collect::<Result<Vec<NodeId>, _>>()?;
+        let endpoints = res.endpoints.into_iter()
+            .map(|e| {
+                e.address.parse().map(|ip: IpAddr| SocketAddr::new(ip, e.port as u16))
+            })
+            .collect::<Result<Vec<SocketAddr>, _>>()?;
+
+        Ok(FindNodeResponse {
+            identities,
+            endpoints,
+            seen: res.seen_ts,
+            slot: res.slot,
+            encryption: res.supported_encryptions,
+        })
+    }
+);
```

Comment:
I would prefer to have better structure for Node information in `ya-relay-client` than to do heavy mapping in macro